zh.perl6intro.adoc

```diff
@@ -2781,74 +2779,74 @@ regex `m/:s Perl 6/` 中的 `:s` 强制考虑空白并且不会被删除。
 
 让我们检查一个邮件是否合法。 +
 我们假设一个合法的电子邮件地址的形式如下: +
-first name [dot] last name [at] company [dot] (com/org/net)
+名 [点] 姓 [at] 公司名 [点] (com/org/net)
 
 WARNING:  这个例子中用于电子邮件检测的 regex 不是很准确。 +
 它的核心意图是用来解释 Perl 6 中的 regex 的功能的。 +
 不要在生产中原样使用它。
 
 [source,perl6]
-.Script
+.脚本
 ----
 my $email = 'john.doe@perl6.org';
 my $regex = / <:L>+\.<:L>+\@<:L+:N>+\.<:L>+ /;
 
 if $email ~~ $regex {
-  say $/ ~ " is a valid email";
+  say $/ ~ " 是一个合法的Email地址";
 } else {
-  say "This is not a valid email";
+  say "这不是合法的Email地址";
 }
 ----
 
 .输出
 
-`john.doe@perl6.org is a valid email`
+`john.doe@perl6.org 是一个合法的Email地址`
 
 .解释
 
 `<:L>`  匹配单个字符 +
 `<:L>+` 匹配单个字符或更多字符 +
-`\.`  匹配单个点号字符 +
-`\@`  匹配单个  [at] 符号 +
+`\.`  匹配单个 . 符号 +
+`\@`  匹配单个 @ 符号 +
 `<:L+:N>` 匹配一个字母或数字 +
 `<:L+:N>+` 匹配多个字母或数字 +
 
 其中的 regex 可以分解成如下:
 
-* *first name* `<:L>+`
+* *名* `<:L>+`
 
-* *[dot]* `\.`
+* *[点]* `\.`
 
-* *last name* `<:L>+`
+* *姓* `<:L>+`
 
 * *[at]* `\@`
 
-* *company name* `<:L+:N>+`
+* *公司名* `<:L+:N>+`
 
-* *[dot]* `\.`
+* *[点]* `\.`
 
 * *com/org/net* `<:L>+`
 
 [source,perl6]
-.可选地, 一个 regex 可以被分解成多个具名 regexes。
+.可选地, 一个 regex 可以被分解成多个具名 regex 。
 ----
 my $email = 'john.doe@perl6.org';
-my regex many-letters { <:L>+ };
-my regex dot { \. };
+my regex 多个字符 { <:L>+ };
+my regex 点 { \. };
 my regex at { \@ };
-my regex many-letters-numbers { <:L+:N>+ };
+my regex 多个字符数字 { <:L+:N>+ };
 
-if $email ~~ / <many-letters> <dot> <many-letters> <at> <many-letters-numbers> <dot> <many-letters> / {
-  say $/ ~ " is a valid email";
+if $email ~~ / <多个字符> <点> <多个字符> <at> <多个字符数字> <点> <多个字符> / {
```

Comment:
many-letters: 多个字母
many-letters-numbers: 多个字母与数字

Reply:
这里是我疏忽了

Reply:
Method 'plot' must be resolved by class combo-chart because it exists in multiple roles (line-chart, bar-chart) 类'combo-chart'的方法'plot'必须被解决，因为存在多个roles（line-chart, bar-chart）
我怀疑这里的resolved应该用“决定”比较好。

匿名函数的表示方法 pointy block 我暂时保留了原文，翻译成“尖块”怎么样？

Reply:
> resolved 是否译可为「解析」呢？
> pointy block 也可以不译，或者译为「尖号块」呢？

第一个问题：这里“解析”倾向于从计算机的角度描述，而“决定”偏向于用户的角度来描述。

第二个问题： “->” 是“尖号”吗？但是“->”这个符号在文中也没有名字。如果是那么可以称为“尖号代码块”。

并且我在搜索“->”的时候发现文中有两个“->”显示错误（在7.5.1）：
<img width="682" alt="image" src="https://user-images.githubusercontent.com/6602710/57081716-e5ea4b00-6d27-11e9-8285-48e75e0c6e5a.png">

很好奇 “resolved” 和 “pointy Block” 这个在 Learning Perl 6 的中文版中是如何翻译的。

Reply:
我会去修改->的显示错误。 

> resolved 是否译可为「解析」呢？
> pointy block 也可以不译，或者译为「尖号块」呢？
因为存在多个role，所以必须决定使用其中哪一个的方法，我是这么想的。
pointy block不译等其他翻译决定了再跟进也好。

Reply:
> 并且我在搜索“->”的时候发现文中有两个“->”显示错误（在7.5.1）：
> <img alt="image" width="682" src="https://user-images.githubusercontent.com/6602710/57081716-e5ea4b00-6d27-11e9-8285-48e75e0c6e5a.png">
我刚好又发现了另一个问题，
Int:D是要求一个已定义的Int
Int:U是要求一个未定义的Int